Contribute/how-to-write-links.md

```diff
@@ -57,42 +77,269 @@ To create an inline link from a Docs technical article to another Docs technical
 
-## Links to anchors
+> In most cases, articles in the same _docset_ have the same `<product-service>` URL fragment. For
+> example:
+> - Same docset
+>   - `https://docs.microsoft.com/dotnet/core/get-started`
+>   - `https://docs.microsoft.com/dotnet/framework/install`
+> - Different docset
+>   - `https://docs.microsoft.com/dotnet/core/get-started`
+>   - `https://docs.microsoft.com/visualstudio/whats-new`
+
+## Bookmark links
+
+For a bookmark link to a heading in the *current* file, use a hash symbol followed by the lowercase
+words of the heading. Remove punctuation from the heading and replace spaces with dashes:
+
+```markdown
+[Managed Disks](#managed-disks) ```
 
-You do not have to create anchors. They're automatically generated at publishing time for all H2 headings. The only thing you have to do is create links to the H2 sections.
+To link to a bookmark heading in another article, use the file-relative or site-relative link plus a
+hash symbol, followed by the words of the heading. Remove punctuation from the heading and replace
+spaces with dashes:
 
-- To link to a heading within the same article:
+```markdown
+[Managed Disks](../../linux/overview.md#managed-disks)
+```
 
-  `[link](#the-text-of-the-H2-section-separated-by-hyphens)`
-  `[Create cache](#create-cache)`
+You can also copy the bookmark link from the URL. To find the URL, hover your mouse over the heading
+line on docs.microsoft.com. You should see a link icon appear:
 
-- To link to an anchor in another article:
+![Link icon in article heading](media/contribute-how-to-write-links/bookmark-link.png)
 
-  `[link text](../directory/article-name.md#anchor-name)`
-  `[Configure your profile](../directory/media-services-create-account.md#configure-your-profile)`
+Click the link icon and then copy the bookmark anchor text from the URL (that is, the part after the
+hash).
 
-## Links from includes
+### Explicit anchor links
+
+Adding explicit anchor links using the `<a>` HTML tag aren't required or recommended, except in hub
+and landing pages. Instead, use the auto-generated bookmarks as described in
+[bookmark links](#bookmark-links). For hub and landing pages, declare anchors as follows:
+
+```markdown
+## <a id="anchortext" />Header text
+```
+
+or
+
+```markdown
+## <a name="anchortext" />Header text
+```
+
+And the following to link to the anchor:
+
+```markdown
+To go to a section on the same page:
+[text](#anchortext)
+
+To go to a section on another page.
+[text](filename.md#anchortext)
+```
+
+> [!NOTE]
+> Anchor text must always be lowercase and not contain spaces.
+
+## XRef (cross reference) links
```

Comment:
Was this section just copied from the internal contributor's guide whole cloth?

Reply:
Yes. My intent was to pull in information that was missing in the public doc, fix some errors, and clarify some guidance.